src/app/action-items/action-items.component.ts

```diff
@@ -14,38 +16,41 @@ export interface ActionItemMapped extends ActionItem {
   templateUrl: './action-items.component.html',
   styleUrls: ['./action-items.component.css']
 })
-export class ActionItemsComponent implements OnInit {
+export class ActionItemsComponent implements OnInit, OnDestroy {
   dataSource: ActionItemMapped[];
-  isloadingActionItems = false;
-  private currentDate: Date = new Date();
+  isLoadingActionItems = false;
+  isLoadingActionItem = false;
+  actionItemId: string | undefined;
+  actionItem: ActionItem;
+  errorMessage: string | undefined;
+  justAddedItemId: string;
+  private querySubscription: Subscription;
+  private itemRequest: Subscription;
+  public ngxScrollToDestination: string;
 
   constructor(
     private tasksService: TasksService,
-    private daysCountService: DaysLeftToDeadlineService,
-    private matDialog: MatDialog
+    private matDialog: MatDialog,
+    private route: ActivatedRoute,
+    private router: Router,
+    private elRef: ElementRef,
+    private scrollToService: ScrollToService
   ) {}
+
   ngOnInit() {
     this.retrieveActionItems();
+    this.getQueryParams();
```

Comment:
`getQueryParams` name suggests that this function will return some query parameters. Instead, it is responsible for subscribing to the query params but it does not return anything. Please provide better name.